### PR TITLE
MODPUBSUB-1 - Create a new module "mod-pubsub"

### PR DESCRIPTION
**Pull request by RuslanLavrov**
* RMB module generated;
* made configuration for log4j2;
* added dockerfile;
* added stub endpoint and test;

[pub-sub RFC](https://github.com/folio-org/rfcs/blob/12c2ffc0da5768ec1e183ba124a83dc75a1662d1/text/0000-pub-sub.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+```
+curl -w '\n' -X POST -D -   \
+    -H "Content-type: application/json"   \
+    -d @target/TenantModuleDescriptor.json \
+    http://localhost:9130/_/proxy/tenants/<tenant_name>/modules
+```
+
+## Issue tracker
+
+See project [MODDATAIMP](https://issues.folio.org/browse/MODDATAIMP)
```
Comment (KaterynaSenchenko) [inline README.md]: set link to modpubsub jira project

File: descriptors/ModuleDescriptor-template.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      "handlers": [
+        {
+          "methods": ["GET"],
+          "pathPattern": "/pubsub"
+        }
+      ]
+    },
+    {
+      "id": "_tenant",
+      "version": "1.0",
```
Comment (KaterynaSenchenko) [inline descriptors/ModuleDescriptor-template.json]: there is a newer version of _tenant api - 1.2

File: src/test/java/org/folio/rest/impl/AbstractRestTest.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  @BeforeClass
+  public static void setUpClass(final TestContext context) throws Exception {
+    Async async = context.async();
+    vertx = Vertx.vertx();
+    port = NetworkUtils.nextFreePort();
+    String okapiUrl = "http://localhost:" + port;
+    PostgresClient.stopEmbeddedPostgres();
+    PostgresClient.closeAllClients();
+    useExternalDatabase = System.getProperty(
+      "org.folio.source.record.manager.test.database",
```
Comment (KaterynaSenchenko) [inline src/test/java/org/folio/rest/impl/AbstractRestTest.java]: property from different module

File: src/test/java/org/folio/rest/impl/AbstractRestTest.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+      "embedded");
+
+    switch (useExternalDatabase) {
+      case "environment":
+        System.out.println("Using environment settings");
+        break;
+      case "external":
+        String postgresConfigPath = System.getProperty(
+          "org.folio.source.record.manager.test.config",
```
Comment (KaterynaSenchenko) [inline src/test/java/org/folio/rest/impl/AbstractRestTest.java]: and here

File: src/test/java/org/folio/rest/impl/AbstractRestTest.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+          "/postgres-conf-local.json");
+        PostgresClient.setConfigFilePath(postgresConfigPath);
+        break;
+      case "embedded":
+        PostgresClient.setIsEmbedded(true);
+        PostgresClient.getInstance(vertx).startEmbeddedPostgres();
+        break;
+      default:
+        String message = "No understood database choice made." +
+          "Please set org.folio.source.record.manager.test.database" +
```
Comment (KaterynaSenchenko) [inline src/test/java/org/folio/rest/impl/AbstractRestTest.java]: and here

File: README.md (hunk 1 above)
Comment (Igor-Gorchakov) [reply README.md]: Please add a link to the google document with design for pub-sub (powered by Taras S.)

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,377 @@
...
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.logging.log4j</groupId>
+      <artifactId>log4j-slf4j-impl</artifactId>
+      <version>2.9.1</version>
+    </dependency>
+    <dependency>
+      <groupId>org.folio</groupId>
+      <artifactId>data-import-utils</artifactId>
```
Comment (Igor-Gorchakov) [inline pom.xml]: Do we really need this dependency here ?

File: README.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
+# mod-sample
```
[the remark this answers is not in the document]
Comment (dcrossleyau) [reply README.md]: mod-pubsub

**Pull request merged**